### PR TITLE
JIT: Clean up some call multi-reg ret handling

### PR DESCRIPTION
**Pull request by jakobbotsch**
- Switch some `IsMultiRegReturnedType` calls to use `GenTreeCall::HasMultiRegRetVal`.
- Remove `Compiler*` parameter of `TreatAsShouldHaveRetBufArg`; the function it was using on `Compiler` is static

### CODE REVIEW

File: src/coreclr/jit/fginline.cpp [hunk 1]
```diff
@@ -405,55 +395,23 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
         // candidates.
         //
         // Do the deferred work now.
-        if (retClsHnd != NO_CLASS_HANDLE)
+        if ((*use)->IsCall() && (*use)->AsCall()->HasMultiRegRetVal())
```
Comment (SingleAccretion) [inline src/coreclr/jit/fginline.cpp:398]: Does this not cause diffs on platforms with multi-reg `LONG` returns?

Comment (jakobbotsch) [reply src/coreclr/jit/fginline.cpp:398]: The 32 bit builds certainly look unhappy.

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @EgorBo 

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603602&view=ms.vss-build-web.run-extensions-tab). Some TP regressions. They come from different inlining decisions in `gtHasReg` because of the switch from `ShouldHaveRetBufArg` to `TreatAsShouldHaveRetBufArg` in `HasMultiRegRetVal`. Hopefully native PGO will smooth that over.

**Comment by jakobbotsch** [comment]
Ping @EgorBo

**Review by EgorBo — APPROVED** [review]

**Pull request merged**